### PR TITLE
Implement Learning Object overview in dashboard side panel

### PR DESCRIPTION
**Pull request by nrwinner**
This PR introduces the UI for a Learning Object's overview in the dashboard's side panel. It also updates all of the spec files in the dashboard component to run and includes them in the circleci config. This will remain feature-toggled until the underlying logic for revisions is available.
screencapture-localhost-4200-onion-dashboard-2019-06-24-14_24_07

### CODE REVIEW

**Comment by nrwinner** [comment]
The tests do pass locally but for whatever reason CircleCI is having a hard time with them. Keeps running out of memory.

**Comment by nvisal1** [comment]
To troubleshoot the Circle CI memory issue, I would recommend running the tests locally with any combination of the following flags.
`--logHeapUsage` -  Outputs heap usage after every test.
`--runInBand` - Runs all tests serially instead of utilizing worker pool of child processes 
`--detectLeaks`- Reports memory leaks
`--expose-gc` - Logs garbage collection
`--detectOpenHandles` - Logs any issue preventing Jest from exiting after test completion

For some reason `--detectLeaks` and `--expose-gc` aren't listed in the docs below. They seem to be popular in thread discussions though. I don't believe that they deprecated. 

Docs: https://jestjs.io/docs/en/cli

File: src/app/onion/dashboard/components/side-panel-content/learning-object/learning-object.component.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div class="learning-object">
+  <div class="learning-object__row">
+    <div tip="{{ learningObject?.name }}" tipDisabled="{{ !learningObject?.name || learningObject?.name.length <= 70 }}" tipLocation="bottom" class="learning-object__name">
```
Comment (pzalep1) [inline src/app/onion/dashboard/components/side-panel-content/learning-object/learning-object.component.html]: tabindex and aria-label

File: src/app/onion/dashboard/components/side-panel-content/learning-object/learning-object.component.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+      {{ (learningObject?.name | slice:0:69) + (learningObject?.name.length > 70 ? '...' : '') }}
+    </div>
+  </div>
+  <div class="learning-object__row learning-object__row--delim">
+    <div [attr.class]="'learning-object__length ' + learningObject.length">{{ learningObject.length | titlecase }}</div>
```
Comment (pzalep1) [inline src/app/onion/dashboard/components/side-panel-content/learning-object/learning-object.component.html]: tabindex and aria-label

File: src/app/onion/dashboard/components/side-panel-content/learning-object/learning-object.component.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    <div class="learning-object__date">{{ learningObject.date | date:'mediumDate' }}</div>
```
Comment (pzalep1) [inline src/app/onion/dashboard/components/side-panel-content/learning-object/learning-object.component.html]: tabindex and aria-label

File: src/app/onion/dashboard/components/side-panel-content/learning-object/learning-object.component.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    <div class="learning-object__collection">{{ learningObject.collection | collection | async }}</div>
```
Comment (pzalep1) [inline src/app/onion/dashboard/components/side-panel-content/learning-object/learning-object.component.html]: tabindex and aria-label

File: src/app/onion/dashboard/components/side-panel-content/side-panel-content.component.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
+<div class="topbar"></div>
+
+<div class="side-panel">
+  <div class="side-panel__object">
+    <clark-panel-learning-object
+      [learningObject]="learningObject"
+    ></clark-panel-learning-object>
+    <div class="side-panel__button-bar">
+      <button
```
Comment (pzalep1) [inline src/app/onion/dashboard/components/side-panel-content/side-panel-content.component.html]: aria-label

File: src/app/onion/dashboard/components/side-panel-content/side-panel-content.component.html [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+          learningObject.author.username,
+          learningObject.name
+        ]"
+      >
+        Visit <i class="far fa-external-link"></i>
+      </button>
+    </div>
+  </div>
+  <div class="side-panel__metrics">
+    <div class="metrics__title title">Metrics</div>
```
Comment (pzalep1) [inline src/app/onion/dashboard/components/side-panel-content/side-panel-content.component.html]: tabindex and aria-label

**Pull request merged**